src/staking/block_validator.cpp

```diff
@@ -194,10 +195,22 @@ class BlockValidatorImpl : public AbstractBlockValidator {
     }
 
     // check that all other transactions are no coinbase transactions
+    CTransactionRef prev_tx = *block.vtx.cbegin();
     for (auto tx = block.vtx.cbegin() + 1; tx != block.vtx.cend(); ++tx) {
       if ((*tx)->GetType() == +TxType::COINBASE) {
         return BlockValidationResult(Error::COINBASE_TRANSACTION_AT_POSITION_OTHER_THAN_FIRST);
       }
+      const int ordering = (*tx)->GetHash().CompareAsNumber(prev_tx->GetHash());
```

Comment:
Some tests to cover the following logic would be really nice to have.

Reply:
The block validator is tested through the LegacyValidationInterface which has a New (goal) and a Legacy (currently in use) implementation. The exact same tests are run on both the new and the old interface, see `validation_tests`. This also tests the transaction ordering.